src/api/spec/models/project_spec.rb

```diff
@@ -287,4 +287,86 @@
       }
     end
   end
+
+  describe '#open_requests' do
+    shared_examples 'with_open_requests' do
+      let(:admin_user) { create(:admin_user, login: 'king') }
+      let(:confirmed_user) { create(:confirmed_user, login: 'confirmed_user') }
+      let(:other_project) { create(:project) }
+      let(:package) { create(:package) }
+      let(:other_package) { create(:package) }
+
+      let!(:review) do
+        create(:review_bs_request, creator: admin_user.login, target_project: project.name, source_project: other_project.name,
+               target_package: package.name, source_package: other_package.name, reviewer: confirmed_user)
+      end
+
+      let!(:target) { create(:bs_request, creator: confirmed_user.login, source_project: project.name) }
+      let!(:other_target) do
+        create(:bs_request_with_submit_action, creator: admin_user.login, target_project: project.name, source_project: other_project.name,
+                                                                          target_package: package.name, source_package: other_package.name)
+      end
+      let!(:declined_target) do
+        create(:declined_bs_request, creator: confirmed_user.login, target_project: other_project.name, source_project: project.name,
+                                                                          target_package: package.name, source_package: other_package.name)
+      end
```

Comment:
hm, on a second thought: these requests are only identified via the source in this situation.

As long as we don't have an interface in webui to search for not existing objects we may be lucky here. The api is offering this though, so we should maybe at a remark at target scope that it must not be used if you search for requests of possible not anymore existing targets.

Are we sure that webui can still show requests where the target object is nil and not just crash btw?